golden/extension.xsd.hs

```diff
@@ -51,6 +51,7 @@ instance FromDom AnyXmlBasetype where
       "Basetype" -> TheXmlBasetype <$> fromDom
       "Ext2" -> TheXmlExt2 <$> fromDom
       "Ext1" -> TheXmlExt1 <$> fromDom
+      _ -> fail "unexpected type"
```

Comment:
This relies on `fail` being in the `Monad` typeclass, which will be removed in the future. It would be better to throw a [`ParserError`](https://hackage.haskell.org/package/dom-parser-3.1.0/docs/Text-XML-DOM-Parser-Types.html#t:ParserError).